src/main/java/pl/javastart/task/DataReadWrite.java

```diff
@@ -0,0 +1,107 @@
+package pl.javastart.task;
+
+import java.io.*;
+import java.util.Scanner;
+
+public class DataReadWrite {
+    private static final int CLOSE_PROGRAM = 0;
+    private static final int ADD_TO_THE_QUEUE = 1;
+    private static final int GET_OUT_OF_THE_QUEUE = 2;
+    private final Scanner sc = new Scanner(System.in);
+    private final MotQueue motQueue = new MotQueue();
+
+    public void motApp() throws IOException {
+        readFile();
+        int option = -1;
+        do {
+            printOptions();
+            option = sc.nextInt();
+            sc.nextLine();
+            if (option == CLOSE_PROGRAM) {
+                if (motQueue.isEmpty()) {
+                    return;
+                } else {
+                    saveQueueToTheFile();
+                }
```

Comment:
Jeżeli modyfikujesz plik przy każdej modyfikacji kolejki, to wydaje się już zbędne zapisywanie stanu przy zamykaniu, aplikacji